src/Microsoft.Health.Dicom.Core/Features/Update/UpdateRequestValidator.cs

```diff
@@ -0,0 +1,88 @@
+using Microsoft.Health.Dicom.Core.Messages.Update;
+using Microsoft.Health.Dicom.Core.Models.Update;
+
+namespace Microsoft.Health.Dicom.Core.Features.Update;
+
+/// <summary>
+/// Provides functionality to validate an <see cref="UpdateInstanceRequest"/>.
+/// </summary>
+public static class UpdateRequestValidator
+{
+    /// <summary>
+    /// Validates an <see cref="UpdateSpecification"/>.
+    /// </summary>
+    /// <param name="updateSpecification">The request to validate.</param>
+    /// <exception cref="BadRequestException">Thrown when request body is missing.</exception>
+    /// <exception cref="UidValidation">Thrown when the specified StudyInstanceUID is not a valid identifier.</exception>
+    public static void ValidateRequest(UpdateSpecification updateSpecification)
+    {
+        EnsureArg.IsNotNull(updateSpecification, nameof(updateSpecification));
+        if (updateSpecification.StudyInstanceUids == null || updateSpecification.StudyInstanceUids.Count == 0)
+        {
+            throw new BadRequestException(string.Format(CultureInfo.CurrentCulture, DicomCoreResource.MissingRequiredField, nameof(updateSpecification.StudyInstanceUids)));
+        }
+        foreach (var StudyInstanceUid in updateSpecification.StudyInstanceUids)
+        {
+            UidValidation.Validate(StudyInstanceUid, nameof(StudyInstanceUid), allowEmpty: true);
+        }
+    }
+
+    /// <summary>
+    /// Validates a <see cref="DicomDataset"/>.
+    /// </summary>
+    /// <param name="dataset">The Dicom dataset to validate.</param>
+    /// <exception cref="BadRequestException">Thrown when dicom tag or value validation fails</exception>
+    public static void ValidateDicomDataset(DicomDataset dataset)
+    {
+        EnsureArg.IsNotNull(dataset, nameof(dataset));
+        List<string> invalidTags = new List<string>();
+        List<string> invalidTagValues = new List<string>();
+        string errorString = string.Empty;
+
+        foreach (DicomItem item in dataset)
+        {
+            if (!UpdateTags.UpdateFilterTags.Contains(item.Tag))
+            {
+                invalidTags.Add(item.Tag.ToString());
+            }
+            else
+            {
+                try
+                {
+                    item.Validate();
+                }
+                catch (DicomValidationException)
+                {
+                    invalidTagValues.Add(item.Tag.ToString());
+                }
+            }
+        }
+        if (invalidTags.Count > 0)
+        {
+            errorString = string.Concat(string.Format(CultureInfo.CurrentCulture, DicomCoreResource.DicomUpdateTagValidationFailed, GetCommaSeparatedTags(invalidTags)), "\n");
+        }
+        if (invalidTagValues.Count > 0)
+        {
+            errorString = string.Concat(errorString, string.Format(CultureInfo.CurrentCulture, DicomCoreResource.DicomUpdateTagValueValidationFailed, GetCommaSeparatedTags(invalidTagValues)));
+        }
```

Comment:
So incase of both invalid tags and invalid values, we are sending either one of them?

Reply:
No, we will send both the tag validation and tag value validation errors together.

Reply:
Sample response would be as below.
```
Updating the following tags are not supported: '(0010,1111)'
Validation failed for the values of following tags: '(0010,0010)'
```